funcframework/framework.go

```diff
@@ -273,6 +273,10 @@ func runUserFunctionWithContext(ctx context.Context, w http.ResponseWriter, r *h
 }
 
 func writeHTTPErrorResponse(w http.ResponseWriter, statusCode int, status, msg string) {
+	// Ensure logs end with a newline otherwise they are grouped incorrectly in SD.
+	if !strings.HasSuffix(msg, "\n") {
```

Comment:
This might lead to empty log lines where there is a `defer fmt.Fprintf(os.Stderr)` above.

Reply:
True, though that defer only fires in the GCF context (when `K_SERVICE` is set).  That said, no caller of this function has a final newline, so it wouldn't be terrible to universally append one on the log message.

Also, maybe it would be better to use `log` rather than `fmt`?

Reply:
I'm wondering if http://cl/211059806 is still required? Seems worth investigating.

Reply:
OK. I suggested we keep the original bug open from that CL. I'm guessing we should move it out of b/ and into GH. Resolving this and will see about attaching another PR to resolve the defer statements issue.